### ISSUE TITLE
[Feature Request]: Add ability to control tooltip's visibility on hover

### ISSUE THREAD

**Issue by ilyas-sov** [carbon-design-system/carbon#17112]
### The problem

Current implementation of the Button's tooltip is that the tooltip stays visible on button hover or the the tooltip itself hover. This behavior is useful if we have interactive content inside. But if the tooltip is for information only like in IconButton component it could overflow the interactive content around and to reach the content around we need to move the mouse out of the tooltip. It is very uncomfortable especially in the lists.

https://github.com/user-attachments/assets/281db691-b048-46cf-830b-96c24899c7e7

### The solution

Create a prop which will control this hover behavior.

### Business priority

Medium Priority = upcoming release but is not pressing

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by guidari** [comment]
@carbon-design-system/design can anyone also weight on this? It seems. to me like a valid use case for an enhancement.

**Comment by tay1orjones** [comment]
@ilyas-sov What you're suggesting used to be the way the component operated in the past. We fixed it, https://github.com/carbon-design-system/carbon/issues/16152, because this is a requirement to pass [SC 1.4.13: Content on Hover or Focus (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html)

> Hoverable
> If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing;

If we added a prop to control this behavior it would be allowing consumers to purposefully fail an accessibility requirement, which isn't something we'd do or recommend. 

Your concern though,

> to reach the content around we need to move the mouse out of the tooltip

can be somewhat solved by the additional thing we fixed in #16152 which is the dismissable requirement. The tooltip can be dismissed without moving the mouse by pressing <kbd>esc</kbd>
